src/main/java/com/mojang/brigadier/tree/CommandNode.java

```diff
@@ -88,8 +88,6 @@ public void addChild(final CommandNode<S> node) {
                 arguments.put(node.getName(), (ArgumentCommandNode<S, ?>) node);
             }
         }
-
-        children = children.entrySet().stream().sorted(Map.Entry.comparingByValue()).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e1, LinkedHashMap::new));
```

Comment:
Notice that the map is comparing by value rather than by key, as seen here:

https://github.com/Mojang/brigadier/blob/9542f80de2685ffad8af653be87e04d508d81c65/src/main/java/com/mojang/brigadier/tree/CommandNode.java#L173-L180

With your patch, the map will mix and match the literal and argument nodes by the node name, which breaks the original contract of the `getChildren` method of returning literal nodes first in the resulting collection.

Imo a better solution would be keep a tree set and a hash map together.

Reply:
Yes it is a behavior change, but one that doesn't seem to matter.

The dispatcher does not iterate this list. arguments and literals are stored in secondary maps and those maps are what is used.

If Mojang wants to make that order matter elsewhere, I'm going to leave it on them to solve that.

Reply:
I strongly oppose your suggestion. Imo you should just keep these nodes in a tree set instead (why do you need the string keys when you can just look up the string keys from the other two maps?) and the iteration order of these nodes is actually important as literal branches are of higher priority in case of conflict with argument branches.

Reply:
No because addChild still looks up by name to merge. That would break stuff.

This change does not break anything as the only place its iterated is findAmbiguities which order is not important there.

I just re-verified this change and can confirm again that switching to a TreeMap here does not break any behavior.

The change as supplied solves the problem and does not break any behavior of the system, so that is all that is needed.

Reply:
My suggestion is to change `children` to a Tree*Set*. Look up by names will simply be look up to two maps instead. Also this is just an implementation detail, it doesn't hurt others.

Since `getChildren` is public api, what works for paper spigot does not necessarily work out well for other communities, like mojang or other modding communities. Changing this breaks users.

Reply:
Also another step back, since you've said messing up iteration order "does not break any behavior", you should've just removed this sorting logic in its entirety for the best result.

Reply:
I meant iteration order of params vs types. Iteration order was not a defined contract on getChildren. It's a Collection<CommandNode<S>>, clearly indicating no expectations of order.

Regardless, Mojang can take it or leave it. I'm not willing to spend any more time debating this.